pkgs/by-name/ch/chadwm/package.nix

```
@@ -0,0 +1,42 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, xorg
+, imlib2
+, picom
+, feh
+, acpi
+, rofi
+}:
+
+stdenv.mkDerivation {
+  name = "chadwm";
+  version = "0-unstable-2024-02-14";
+  src = fetchFromGitHub {
+    owner = "siduck";
+    repo = "chadwm";
+    rev = "566de97abf2c06a18abec7c776a9d2d2edd12055";
+    hash = "sha256-WX1vsbivOCEduAmTfXNTEKzNHQuh62LVkr5SmYrfte0=";
+  };
+  buildInputs = [
+    imlib2
+    picom
+    feh
+    acpi
+    rofi
+    xorg.libX11
+    xorg.libXft
+    xorg.libXinerama
+    xorg.xsetroot
+  ];
```

```suggestion

  buildInputs = [
    imlib2
    libX11
    libXft
    libXinerama
  ];
  
```

xorg dependencies are passed in via `callPackage`, you can use them directly without the `xorg.` prefix.

Also, I don't believe some of these are necessary, looking at the repo.
* https://github.com/siduck/chadwm/blob/main/chadwm/dwm.c#L23-L47
* https://github.com/siduck/chadwm/blob/main/chadwm/drw.c#L2-L7

---

they are added as dependancies based on [the "other requirments" section of the README file](https://github.com/siduck/chadwm?tab=readme-ov-file#other-requirements).

there's more to it than just building dwm.c

---

Those deps are binaries, they are doing nothing here.

They're only used in some auxiliary scripts, which aren't packaged.

---

hmm, it seems that is correct, i wish there was a way to package those scripts, but that would be in the domain of homeManager as far as i can tell.